src/tye_lab_to_nwb/ast_ecephys/ast_ecephysnwbconverter.py

```diff
@@ -0,0 +1,65 @@
+from typing import Optional
+
+from pynwb import NWBFile
+
+from neuroconv import NWBConverter
+from neuroconv.datainterfaces import (
+    OpenEphysRecordingInterface,
+    PlexonSortingInterface,
+    SLEAPInterface,
+    VideoInterface,
+)
+from tye_lab_to_nwb.ast_ecephys.ast_sortinginterface import AstSortingInterface
+from tye_lab_to_nwb.neurotensin_valence.behavior import NeurotensinEventsInterface
+
+
+class AStEcephysNWBConverter(NWBConverter):
+    """Primary conversion class for the ASt electrophysiology dataset."""
+
+    data_interface_classes = dict(
+        Behavior=SLEAPInterface,
+        Video=VideoInterface,
+        Events=NeurotensinEventsInterface,
+        Recording=OpenEphysRecordingInterface,
+        Sorting=PlexonSortingInterface,
+        FilteredSorting=AstSortingInterface,
+    )
+
+    def get_metadata(self):
+        metadata = super().get_metadata()
+
+        start_times = []
+        for interface in self.data_interface_objects.values():
+            interface_metadata = interface.get_metadata()
+            if "NWBFile" not in interface_metadata:
+                continue
+            if "session_start_time" in interface_metadata["NWBFile"]:
+                start_times.append(interface_metadata["NWBFile"]["session_start_time"])
+
+        # Use the earliest session_start_time
+        metadata["NWBFile"].update(session_start_time=min(start_times))
```

Comment:
If deciding the session start time programmatically like this, how do you guarantee each interface is properly aligned?

Reply:
That is, what if in one session the minimal start time comes from the recording, and in another it comes from the behavior.

Can we just assume they have pre-aligned everything to whichever is first? Or do they instead always align to the recording, for example

Reply:
@CodyCBakerPhD Actually this is because of the Plexon interface, since it was used as an offline sorter, the date it was created is a day after the OpenEphys.